### PR TITLE
Split out CTerm manipulation functionalities of KCFGExplore

### PR DESCRIPTION
**Pull request by ehildenb**
`KCFGExplore` class has grown quite a bit, and broadly contains two types of things: (i) wrappers around the `KoreClient` for the `CTerm` class, and (ii) functionalities for manipulating KCFGs using said wrappers. This PR pulls the first category out into a new submodule of the `cterm` submodule.

- Module `cterm/symbolic.py` is added, with class `CTermSymbolic`, which wraps a `KoreClient` and lifts its functionalities to from Kore to `CTerm` level.
  - The class `CTermSymbolic` doesn't need access to a `KPrint`, instead just a `KDefinition` and a `KompiledKore`.
  - The class `CTermSymbolic` implements two new methods `CTermSymbolic.kast_to_kore(KInner) -> Pattern` and `CTermSymbolic.kore_to_kast(Pattern) -> KInner` as wrappers around the methods in `konvert` which have the definitions applied, and uses them in the various methods instead. In particular, that means we do _not_ have fallback to `kast` binary for these methods (instead crashes), but these codepaths are very well tested now.
- The methods from `KCFGExplore.cterm_*` are moved to the `CTermSymbolic` class, with the prefix `cterm_` dropped.
- The `KCFGExplore` is updated to use this class instead of a `KoreClient` directly, and various `*Prover` classes are updated as well.
- Updates to the testing harness are made:
  - A new fixture `CTermSymbolic` is added, for when only that is needed.
  - The directory `src/tests/integration/kcfg` is moved to `src/tests/integration/cterm`, and lowered to `CTermSymbolic` instead of `KCFGExplore`.

### CODE REVIEW

File: src/pyk/testing/_kompiler.py [hunk 1]
```diff
@@ -226,20 +232,32 @@ def kore_client(self, _kore_server: KoreServer, bug_report: BugReport) -> Iterat
             yield client
 
 
-class KCFGExploreTest(KoreClientTest, KPrintTest):
+class CTermSymbolicTest(KoreClientTest, KompiledTest):
```
Comment (tothtamas28) [inline src/pyk/testing/_kompiler.py:235]: `KoreClientTest` extends `KompiledTest`:

```suggestion
class CTermSymbolicTest(KoreClientTest):
```

**Review by tothtamas28 — APPROVED** [review]

File: src/pyk/testing/_kompiler.py (hunk 1 above)
Comment (ehildenb) [reply src/pyk/testing/_kompiler.py:235]: Done.

File: src/pyk/testing/_kompiler.py [hunk 2]
```diff
@@ -226,20 +232,32 @@ def kore_client(self, _kore_server: KoreServer, bug_report: BugReport) -> Iterat
...
+    @pytest.fixture
+    def cterm_symbolic(
+        self,
+        kore_client: KoreClient,
+        definition: KDefinition,
+        kompiled_kore: KompiledKore,
+        bug_report: BugReport | None,
+    ) -> Iterator[CTermSymbolic]:
+        yield CTermSymbolic(kore_client, definition, kompiled_kore)
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply src/pyk/testing/_kompiler.py]: Done.

File: src/pyk/cterm/symbolic.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        self._kore_client = kore_client
+        self._definition = definition
+        self._kompiled_kore = kompiled_kore
+        self._trace_rewrites = trace_rewrites
+
+    def kast_to_kore(self, kinner: KInner) -> Pattern:
+        return kast_to_kore(self._definition, self._kompiled_kore, kinner, sort=GENERATED_TOP_CELL)
+
+    def kore_to_kast(self, pattern: Pattern) -> KInner:
+        return kore_to_kast(self._definition, pattern)
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply src/pyk/cterm/symbolic.py:59]: New issue to describe it: https://github.com/runtimeverification/pyk/issues/918

**Pull request merged**